src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -312,14 +312,16 @@ private int GetNullCount(long startIndex, int numberOfRows)
 
         protected internal override Apache.Arrow.Array AsArrowArray(long startIndex, int numberOfRows)
         {
+            if (numberOfRows == 0)
+                return new StringArray.Builder().Build();
```

Comment:
Instead of using `StringArray.Builder()`, which will allocate a bunch of new array, how about just:

```C#
return new StringArray(numberOfRows, ArrowBuffer.Empty , ArrowBuffer.Empty , ArrowBuffer.Empty);
```

which won't allocate a bunch of unnecessary memory.